app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt

```diff
@@ -1,29 +1,92 @@
+import org.oppia.app.recyclerview.BindableAdapter
 import org.oppia.app.viewmodel.ViewModelProvider
+import org.oppia.domain.profile.ProfileManagementController
 import javax.inject.Inject
 
 /** The presenter for [ProfileChooserFragment]. */
 @FragmentScope
 class ProfileChooserFragmentPresenter @Inject constructor(
   private val fragment: Fragment,
-  private val viewModelProvider: ViewModelProvider<ProfileChooserViewModel>
+  private val viewModelProvider: ViewModelProvider<ProfileChooserViewModel>,
+  private val profileManagementController: ProfileManagementController
 ) {
+  /** Binds ViewModel and sets up RecyclerView Adapter. */
+  @ExperimentalCoroutinesApi
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
     val binding = ProfileChooserFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
     binding.apply {
       viewModel = getProfileChooserViewModel()
+      lifecycleOwner = fragment
+    }
+    binding.profileRecyclerView.apply {
+      adapter = createRecyclerViewAdapter()
     }
     return binding.root
   }
 
   private fun getProfileChooserViewModel(): ProfileChooserViewModel {
     return viewModelProvider.getForFragment(fragment, ProfileChooserViewModel::class.java)
   }
+
+  @ExperimentalCoroutinesApi
+  private fun createRecyclerViewAdapter(): BindableAdapter<ProfileChooserModel> {
+    return BindableAdapter.Builder
+      .newBuilder<ProfileChooserModel>()
+      .registerViewTypeComputer { value ->
+        value.modelTypeCase.number
+      }
+      .registerViewDataBinderWithSameModelType(
+        viewType = ProfileChooserModel.PROFILE_FIELD_NUMBER,
+        inflateDataBinding = ProfileChooserProfileViewBinding::inflate,
+        setViewModel = this::bindProfileView)
+      .registerViewDataBinderWithSameModelType(
+        viewType = ProfileChooserModel.ADDPROFILE_FIELD_NUMBER,
+        inflateDataBinding = ProfileChooserAddViewBinding::inflate,
+        setViewModel = this::bindAddView)
+      .build()
+  }
+
+  @ExperimentalCoroutinesApi
+  private fun bindProfileView(binding: ProfileChooserProfileViewBinding, data: ProfileChooserModel) {
+    binding.viewModel = data
+    binding.root.setOnClickListener {
+      profileManagementController.loginToProfile(data.profile.id).observe(fragment, Observer {
+        if (it.isSuccess()) {
+          fragment.requireActivity().startActivity(Intent(fragment.context, HomeActivity::class.java))
+        }
+      })
+    }
+  }
+
+  private fun bindAddView(binding: ProfileChooserAddViewBinding, data: ProfileChooserModel) {
+    binding.root.setOnClickListener {
+      if (getAdminAuthFragment() == null) {
+        fragment.requireActivity().supportFragmentManager.beginTransaction()
+          .setCustomAnimations(R.anim.slide_up, R.anim.slide_down, R.anim.slide_up, R.anim.slide_down).add(
+          R.id.profile_chooser_fragment_placeholder,
+          AdminAuthFragment()
+        ).addToBackStack(null).commit()
```

Comment:
Can we use an activity for this, instead, to avoid backstack management? Or, can we just set up the fragment placeholder to be on top of all other views in this activity and use a regular commitNow() + set the placeholder to visible?

Reply:
Both AdminAuthFragment and AddProfileFragment are changed to activities in the next PR.